share/spack/setup-env.sh

```diff
@@ -58,38 +58,38 @@
 
 function spack {
     # Zsh does not do word splitting by default, this enables it for this function only
-    if [ -n "${ZSH_VERSION:-}" ]; then
+    if [[ "${ZSH_VERSION:-}" ]]; then
```

Comment:
@adamjstewart : In `[[ ... ]]` quoting of variables isn't necessary. (In `[ ... ]` it is!)